src/libmeasurement_kit/ooni/orchestrate.cpp

```diff
@@ -29,10 +29,102 @@ std::string production_events_url() {
 std::string testing_events_url() { return MK_OONI_TESTING_PROTEUS_EVENTS_URL; }
 
 /*
+ * Auth
+ */
+
+/*static*/ std::string Auth::make_password() {
+    return mk::random_printable(64);
+}
+
+Error Auth::load(const std::string &filepath) noexcept {
+    ErrorOr<std::string> maybe_data = slurp(filepath);
+    if (!maybe_data) {
+        return maybe_data.as_error();
+    }
+    return loads(*maybe_data);
+}
+
+Error Auth::loads(const std::string &s) noexcept {
+    return json_parse_process_and_filter_errors(s, [&](auto json) {
```

Comment:
What happens if some key is missing or the json document is malformed? How will the user learn of an error?